### DISCUSSION TITLE
WebSocket transport not available. Install simple-websocket for improved performance.

### DISCUSSION THREAD

**Discussion by liuqijie6** [miguelgrinberg/Flask-SocketIO#1873] [Q&A]
I get a warning when I run it. WebSocket transport not available. Install simple-websocket for improved performance.

python=3.8
Flask-SocketIO == 5.2.0

app.py
![image](https://user-images.githubusercontent.com/52588515/189016846-25f83dd9-04f3-4453-8537-7613a2a81a4d.png)

extensions.py
![image](https://user-images.githubusercontent.com/52588515/189016943-6938ecfa-dea4-4856-a6c7-fd51be935fc8.png)

When i use async_mode=gevent： WARNING in __init__: WebSocket transport not available. Install gevent-websocket for improved performance.

When i use async_mode=eventlet： no warning。

I don`t know why.

**Comment by miguelgrinberg** [comment]
I don't understand what you are asking. Depending on how you configure the Socket.IO server you may get warnings regarding things that are missing that can help with performance. Eventlet does not need anything. Gevent does, and threading does as well.
